shared/src/commonMain/kotlin/kmp/shared/extension/Store.kt

```diff
@@ -0,0 +1,70 @@
+package kmp.shared.extension
+
+import kmp.shared.base.ErrorResult
+import kmp.shared.base.ErrorResultException
+import kmp.shared.base.Result
+import kmp.shared.base.error.domain.CommonError
+import kmp.shared.infrastructure.model.UserDto
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.filterNot
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.map
+import org.mobilenativefoundation.store.store5.MutableStore
+import org.mobilenativefoundation.store.store5.StoreReadRequest
+import org.mobilenativefoundation.store.store5.StoreReadResponse
+import org.mobilenativefoundation.store.store5.StoreWriteResponse
+
+fun <Key : Any, Output : Any> MutableStore<Key, Output>.getResultFlow(key: Key): Flow<Result<Output>> =
+    stream<UserDto>(StoreReadRequest.cached(key, refresh = true))
+        .filterNot { it is StoreReadResponse.Loading || it is StoreReadResponse.NoNewData }
+        .map(StoreReadResponse<Output>::toResult)
+
+suspend fun <Key : Any, Output : Any> MutableStore<Key, Output>.getResult(key: Key): Result<Output> =
+    getResultFlow(key).first()
+
+inline fun <T : Any> StoreReadResponse<T>.toResult(): Result<T> = when (this) {
+    is StoreReadResponse.Data -> Result.Success(value)
+
+    is StoreReadResponse.Error.Exception -> Result.Error(CommonError.UnhandledExceptionError(error))
+    is StoreReadResponse.Error.Message -> Result.Error(
+        CommonError.UnhandledExceptionError(RuntimeException(message)),
+    )
+
+    is StoreReadResponse.Error.Custom<*> -> {
+        val errorResult: ErrorResult = error as? ErrorResult ?: error("Unknown custom error $error")
+        Result.Error(errorResult)
+    }
+
+    else -> error("Illegal store state $this. Should be filtered out")
```

Comment:
And here as well

Reply:
But one of the possible values there is "Loading". If you don't filter it out in you implementation, you'd get errors when something would be loading. I'd think that crashing is the better option here since you should see it immediately when debugging.   
The `NoNewData` is different question, that could probably be and error although I am not sure when is it used. The documentation says:
> No new data event dispatched by Store to signal the Fetcher returned no data (i.e., the returned kotlinx.coroutines.flow.Flow, when collected, was empty)

Reply:
What would you suggest I return? We don't have "unknown error" and honestly I am not fan of even having unknown error.